### DISCUSSION TITLE
Kafka ui support the confluent kafka

### DISCUSSION THREAD

**Discussion by mullasaddam** [provectus/kafka-ui#2906] [Q&A] [answered]
Im try to create the Kafka ui for confluent kafka But our applicaion have username and password .
How to pass the user name and password in kafkaui .For Kafka ui i'm using simple docker file .

I'm passing username and password in docker in env section 
 KAFKA_CLUSTERS_0_PROPERTIES_SASL_USERNAME=xxxxx
 KAFKA_CLUSTERS_0_PROPERTIES_SASL_PASSWORD=yyyyy

even i'm getting error 
Failed to collect cluster Confluent-Kafka-Dev info
 java.lang.IllegalStateException: Error while creating AdminClient for Cluster Confluent-Kafka-Dev
     at com.provectus.kafka.ui.service.AdminClientServiceImpl.lambda$createAdminClient$3(AdminClientServiceImpl.

**Comment by Haarolean** [comment, marked as answer]
Hi,

most likely you'll have to use SASL auth.
Please take a look at [this example](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/kafka-ui-sasl.yaml#L17).

**Reply by mullasaddam** [reply]
i'm using below docker file for kafka ui for confluent kafka application . in my application i have user name and password .But facing same error please 

error screenshot:-

ERROR [parallel-2] c.p.k.u.s.StatisticsService: **Failed to collect cluster dev_aws_us_west_2 info
kafka-ui    | java.lang.IllegalStateException: Error while creating AdminClient for Cluster** 

version: '3.4'
services:
  kafka-ui:
    image: provectuslabs/kafka-ui:master
    container_name: xx-you
    ports:
      - "xxx:yyy"
    restart: always
    environment:
      KAFKA_CLUSTERS_0_NAME: xxxxx
      KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS: xxxxxx
      KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL: SASL_SSL
      KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM: PLAIN
      KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG: 'org.apache.kafka.common.security.plain.PlainLoginModule required username="xxxxxx" password="xxxxxxxxxx";'

**Reply by mullasaddam** [reply]
error:- Caused by: org.apache.kafka.common.errors.ClusterAuthorizationException: Cluster authorization failed.

**Reply by iliax** [reply]
@mullasaddam I checked today, following config works fine 
```
docker run -p 8080:8080 \
        -e KAFKA_CLUSTERS_0_NAME=confluent-test \
        -e KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS={endpoint}:9092 \
        -e KAFKA_CLUSTERS_0_DISABLELOGDIRSCOLLECTION=true \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL=SASL_SSL \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=PLAIN \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG='org.apache.kafka.common.security.plain.PlainLoginModule required username="XXXXX" password="XXXXXX";' \
        -e KAFKA_CLUSTERS_0_SCHEMAREGISTRY=https://XXXXX.aws.confluent.cloud \
        -e KAFKA_CLUSTERS_0_SCHEMAREGISTRYAUTH_USERNAME=XXXXX \
        -e KAFKA_CLUSTERS_0_SCHEMAREGISTRYAUTH_PASSWORD=XXXXXX 
        provectuslabs/kafka-ui:latest
```

Note: make sure `KAFKA_CLUSTERS_0_DISABLELOGDIRSCOLLECTION=true` is set

**Reply by mullasaddam** [reply]
still facing same issue .one thing i want to say . i doesn't have any any schema registry . just i want to conform with you above schema registry are from confluent cloud we need to create in confluent kafka ?.

**Reply by iliax** [reply]
@mullasaddam within config above I was connected to SchemaRegistry created in Confluent Cloud. If you not using SR, you can rm `KAFKA_CLUSTERS_0_SCHEMAREGISTRY*` props. 
config will be like 
```
docker run -p 8080:8080 \
        -e KAFKA_CLUSTERS_0_NAME=confluent-test \
        -e KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS=XXXX.aws.confluent.cloud:9092 \
        -e KAFKA_CLUSTERS_0_DISABLELOGDIRSCOLLECTION=true \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL=SASL_SSL \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=PLAIN \
        -e KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG='org.apache.kafka.common.security.plain.PlainLoginModule required username="XXXX" password="XXXX";' \
        provectuslabs/kafka-ui:latest
```

**Reply by Haarolean** [reply]
@mullasaddam "ClusterAuthorizationException" means that you don't have the required permissions granted to your user. No other options here. Check [this](https://github.com/provectus/kafka-ui/wiki/Common-problems#cluster-authorization-failed).

**Reply by mullasaddam** [reply]
just now i change the permissions of user .it will working fine @Haarolean. Thanks for everyone for your support and guidance

**Reply by Haarolean** [reply]
@mullasaddam you're most welcome :)
